datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala

```diff
@@ -0,0 +1,181 @@
+package za.co.absa.hermes.datasetComparison
+
+import com.typesafe.config.{Config, ConfigFactory}
+import org.apache.spark.sql._
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{StructField, StructType}
+import za.co.absa.hermes.utils.HelperFunctions
+
+object DatasetComparisonJob {
+  private val conf: Config = ConfigFactory.load
+  private val errorColumnName: String = conf.getString("dataset-comparison.errColumn")
+  private val tmpColumnName: String = conf.getString("dataset-comparison.tmpColumn")
+  private val comparisonUniqueId: String = conf.getString("dataset-comparison.comparisonUniqueId")
+  private val actualPrefix: String = conf.getString("dataset-comparison.actualPrefix")
+  private val expectedPrefix: String = conf.getString("dataset-comparison.expectedPrefix")
+
+  def main(args: Array[String]): Unit = {
+    val cmd = DatasetComparisonConfig.getCmdLineArguments(args)
+    implicit val sparkSession: SparkSession = SparkSession.builder()
+      .appName(s"Dataset comparison - '${cmd.newPath}' and '${cmd.refPath}'")
+      .getOrCreate()
+
+    execute(cmd)
+  }
+
+  /**
+    * Execute the comparison
+    *
+    * @param cmd Provided configuration for the comparison
+    * @param sparkSession Implicit spark session
+    */
+  def execute(cmd: DatasetComparisonConfig)(implicit sparkSession: SparkSession): Unit = {
+    val dfReader: DataFrameReader = DataFrameReaderFactory.getDFReaderFromCmdConfig(cmd)
+    val expectedDf: DataFrame = dfReader.load(cmd.refPath)
+    val actualDf: DataFrame = dfReader.load(cmd.newPath)
+    val expectedSchema: StructType = getSchemaWithoutMetadata(expectedDf.schema)
+    val actualSchema: StructType = getSchemaWithoutMetadata(actualDf.schema)
```

Comment:
I'd remove all of the explicit types here.

Reply:
I am not sure in the `StructType` cases - the type there is not that obvious IMHO

Reply:
I removed them. It is true that the names are descriptive enough for them not to be needed.